scripts/files/fs_s3.py

```diff
@@ -67,6 +67,96 @@ def read(path: str, needs_credentials: bool = False) -> bytes:
     return file
 
 
+def rename(path: str, new_path: str, needs_credentials: bool = False) -> None:
+    """'Rename' an s3 object: copy an existing object to another with the new key name and delete the old object.
+        The objects have to be in the same bucket.
+
+    Args:
+        path (str): the path of the object to rename.
+        new_path (str): the path with the new name.
+        needs_credentials (bool, optional): Tells if credentials are needed. Defaults to False.
+
+    Raises:
+        Exception: an exception if the path are not in the same bucket
+        ce: a client exception
+        nsb: a no such bucket exception
+        nsk: a no such key exception
+    """
+    start_time = time_in_ms()
+    src_s3_path = parse_path(path)
+    dst_s3_path = parse_path(new_path)
+
+    if src_s3_path.bucket != dst_s3_path.bucket:
+        get_log().error(
+            "rename_s3_different_buckets",
+            path=path,
+            destination=new_path,
+            error="The source and destination path are not in the same bucket",
+        )
+        raise Exception(f"Files {path} and {new_path} are not on the same S3 bucket.")
+
+    s3 = boto3.resource("s3")
+    try:
+        if needs_credentials:
+            s3 = get_session(path).client("s3")
+
+        dst_s3_object = s3.Object(dst_s3_path.bucket, dst_s3_path.key)
+        src_s3_object = s3.Object(src_s3_path.bucket, src_s3_path.key)
+        try:
+            # check if the original file exists
+            src_s3_object.load()
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "404":
+                get_log().error(
+                    "rename_s3_not_exists",
+                    path=path,
+                    destination=new_path,
+                    error="The file to rename does not exist.",
+                )
+            raise Exception(f"{path} does not exists.") from e
+
+        try:
+            # check if the destination file already exists
+            dst_s3_object.load()
+            get_log().error(
+                "rename_s3_already_exists",
+                path=path,
+                destination=new_path,
+                error="A file for the destination path already exists.",
+            )
+            raise Exception(f"{new_path} already exists. The file can't be renamed with this path.")
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "404":
+                # OK: the destination file does not already exists
+                dst_s3_object.copy({"Bucket": src_s3_path.bucket, "Key": src_s3_path.key})
+                # delete the source
+                src_s3_object = s3.Object(src_s3_path.bucket, src_s3_path.key)
```

Comment:
why is the source object recreated here? 

is it not the same object as https://github.com/linz/topo-imagery/pull/195/files#diff-3874c4d626707da994208cc5bced4772a5d0506c9b9912a63ff645f0b05450e7R104

Reply:
I forgot to remove that as I did push a commit later to check if the source object exists